ddtrace/contrib/langchain/patch.py

```diff
@@ -397,10 +397,36 @@ def traced_chat_model_generate(langchain, pin, func, instance, args, kwargs):
         for message_set_idx, message_set in enumerate(chat_completions.generations):
             for idx, chat_completion in enumerate(message_set):
                 if integration.is_pc_sampled_span(span):
-                    span.set_tag_str(
-                        "langchain.response.completions.%d.%d.content" % (message_set_idx, idx),
-                        integration.trunc(chat_completion.text),
-                    )
+                    text = chat_completion.text
+                    message = chat_completion.message
+                    # tool calls aren't available on this property for legacy chains
+                    tool_calls = getattr(message, "tool_calls", None)
+
+                    if text:
+                        span.set_tag_str(
+                            "langchain.response.completions.%d.%d.content" % (message_set_idx, idx),
+                            integration.trunc(chat_completion.text),
+                        )
+                    if tool_calls:
+                        if not isinstance(tool_calls, list):
+                            tool_calls = [tool_calls]
+                        for tool_call_idx, tool_call in enumerate(tool_calls):
+                            span.set_tag_str(
+                                "langchain.response.completions.%d.%d.tool_calls.%d.id"
+                                % (message_set_idx, idx, tool_call_idx),
+                                integration.trunc(str(tool_call.get("id", ""))),
```

Comment:
In the context of a tool call we only consider the output arguments as the completion, meaning we shouldn't need to truncate the completion/tool ID/name

Reply:
oop makes sense 😅